pkg/await/apps_deployment.go

```diff
@@ -503,6 +542,30 @@ func (dia *deploymentInitAwaiter) changeTriggeredRollout() bool {
 	return len(fields) > 0
 }
 
+func (dia *deploymentInitAwaiter) checkPersistentVolumeClaimStatus() {
+	inputs := dia.config.currentInputs
+
+	glog.V(3).Infof("Checking PersistentVolumeClaims status for Deployment %q", inputs.GetName())
+
+	allPVCsReady := true
+	for _, pvc := range dia.pvcs {
+		phase, hasConditions := openapi.Pluck(pvc.Object, "status", "phase")
+		if !hasConditions {
+			return
+		}
+
+		// Success only occurs when there are no PersistentVolumeClaims
+		// defined, or when all PVCs have a status of 'Bound'
+		if phase != statusBound {
+			allPVCsReady = false
+			message := fmt.Sprintf("PersistentVolumeClaim: [%s] is not ready. status.phase currently at: %s", pvc.GetName(), phase)
```

Comment:
We've historically tried to break these up into steps to make it seem sequential (_e.g._, `"[1/2] Waiting for app ReplicaSet be marked available"`). I'm curious to hear your thoughts on whether we should be doing that here, too? Where might it fit in?

Reply:
Hmm great point. Thinking about this more, many other API resources can be embedded within a Deployment other than a PVC: e.g. a ConfigMap, Secret etc. This tells me we should go back to the drawing board to think on how to know of all dependent pieces of a Deployment up front, so that we can track each object required, as well as programmatically form the progress message (e.g. `[1/2]`...`[2/5]` etc.) accordingly to the total number of objects we're waiting on.

Given this, it seems out of scope to hard-code a number of objects for PVCs in this PR, so I'll file this as a separate issue to track.